Lib/pyclbr.py

```diff
@@ -58,31 +59,36 @@ def __init__(self, module, name, file, lineno, parent):
         self.lineno = lineno
         self.parent = parent
         self.children = {}
+        if parent is not None:
+            parent._addchild(name, self)
 
     def _addchild(self, name, obj):
         self.children[name] = obj
 
 
 class Function(_Object):
     "Information about a Python function, including methods."
-    def __init__(self, module, name, file, lineno, parent=None):
-        _Object.__init__(self, module, name, file, lineno, parent)
-
+    def __init__(self, module, name, file, lineno, parent=None, is_async=False):
+        super().__init__(module, name, file, lineno, parent)
+        self.is_async = is_async
+        if isinstance(parent, Class):
+            parent._addmethod(name, lineno)
```

Comment:
I believe my previous comments about addchild and offer to make the change apply to addmethod.

```suggestion
        if isinstance(parent, Class):
            parent._methods[name] = lineno
```